pandas/core/arrays/interval.py

```diff
@@ -1074,14 +1074,24 @@ def overlaps(self, other):
             msg = '`other` must be Interval-like, got {other}'
             raise TypeError(msg.format(other=type(other).__name__))
 
+        def is_empty(intv):
+            return intv.left == intv.right and intv.closed != 'both'
+
+        if is_empty(other):
+            return np.zeros(len(self), dtype=bool)
+
+        empty = (self.left == self.right) & (self.closed != 'both')
+
         # equality is okay if both endpoints are closed (overlap at a point)
+        # and neither is empty.
         op1 = le if (self.closed_left and other.closed_right) else lt
         op2 = le if (other.closed_left and self.closed_right) else lt
 
         # overlaps is equivalent negation of two interval being disjoint:
         # disjoint = (A.left > B.right) or (B.left > A.right)
         # (simplifying the negation allows this to be done in less operations)
-        return op1(self.left, other.right) & op2(other.left, self.right)
+        result = (op1(self.left, other.right) & op2(other.left, self.right))
```

Comment:
small nitpick: I don't think we need the outer parenthesis

Reply:
fixed.